package-system/mcpp/get_and_build_mcpp.py

```diff
@@ -122,6 +132,43 @@ def extract_tarfile(temp_folder):
         return False
 
 
+def init_git_repo(temp_folder):
+    """
+    Runs 'git init' on temp_folder.
+    This is useful to 'git apply' won't fail silently when executing apply_patch(...)
+    REMARK:
+    1- It is very important to set the local git project
+    to not autoconvert LF to CRLF because it causes the patching
+    to fail as 'git apply' is very picky about that.
+    2- You may notice that there's a .gitattributes file that makes sure
+    the patch file remains with LF when fetched from the repo.
+    3- It was also found that 'git apply' also failed if the patch had CRLF
+    AND the local git repo also had CRLF.
+    """
+    pristine_source_path = str((temp_folder / SOURCE_NAME).resolve())
+    git_cmds = [
+        ['git', 'init'],
```

Comment:
FYI: Since you are running the execute_cmd with shell=True, you do not have to separate the arguments out into an array. The execute_cmd will just join them into a singular statement because thats how executing a process through a shell works. 

So git_cmds can just look like:

```
git_cmds = [
   'git init',
   'git config --local core.eol lf',
   'git config --local core.autocrlf false',
...
etc
```

Reply:
Thanks for the tip, although it's the same code from "apply_patch()".